frontend/public/components/app.jsx

```diff
@@ -261,7 +262,7 @@ const App = (props) => {
           <CloudShell />
           <GuidedTour />
         </div>
-        <div id="modal-container" role="dialog" aria-modal="true" />
+        <div id="modal-container" role="dialog" aria-modal="true" aria-label={t('public~Modal')} />
```

Comment:
Although this fixes the accessibility violation, is there a way to improve the semantics of the aria-label? See this [example from MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal), where the aria label links to the modal title, and the description is also linked by aria attributes:

```html
<div id="backdrop" class="no-scroll">
  <div
    role="alertdialog"
    aria-modal="true"
    aria-labelledby="dialog_label"
    aria-describedby="dialog_desc">
    <h2 id="dialog_label">Confirmation</h2>
    <div id="dialog_desc">
      <p>Are you sure you want to delete this file?</p>
    </div>
    <button type="button" onclick="closeDialog(this)">
      No. Close this popup.
    </button>
    <button type="button" onclick="deleteFile(this)">
      Yes. Delete the file.
    </button>
  </div>
</div>
```

Reply:
Good suggestion. 

I don't think we're able to easily do that given we don't know what contents are going to be provided.  See https://github.com/openshift/console/blob/d149ebc6027f145adfebb51be12e96d73a4b75d2/frontend/public/components/factory/modal.tsx#L51 and https://github.com/openshift/console/blob/d149ebc6027f145adfebb51be12e96d73a4b75d2/frontend/packages/console-shared/src/components/modal/Modal.tsx#L15

Reply:
Good point! Seems like `console-shared/src/components/modal/Modal.tsx` is only used in one here, so the usage of 

https://github.com/openshift/console/blob/d149ebc6027f145adfebb51be12e96d73a4b75d2/frontend/packages/console-shared/src/components/catalog/details/CatalogDetailsModal.tsx#L67

could be modified so that the component is no longer needed

For `public/components/factory/modal.tsx`, seems like all usages of `ModalWrapper` and all the usages of `createModalLauncher` I checked make use of the `ModalTitle` component.

I think that as long as there is only one `ModalTitle` present in the DOM at all times, and all usages of the deprecated Modal always use a `ModalTitle`, it should be possible to provide a more accessible name to the modal than just "Modal"

Reply:
/lgtm